server/revision/revision.controller.js

```diff
@@ -2,20 +2,30 @@
 
 const { Revision, User } = require('../shared/database');
 const { resolveStatics } = require('../shared/storage/helpers');
+const revisionService = require('./revision.service');
 
 function index({ repository, query }, res) {
-  const { limit, offset, entity, entityId } = query;
-  const where = { repositoryId: repository.id };
-  if (entity) {
-    where.entity = entity;
-    where.state = { id: entityId };
-  }
+  const { limit, offset, entity, publishedOn } = query;
+  const filter = { ...query, repositoryId: repository.id };
   const include = [{
     model: User,
     paranoid: false,
     attributes: ['id', 'email', 'firstName', 'lastName', 'fullName', 'label']
   }];
-  const opts = { where, include, order: [['createdAt', 'DESC']], limit, offset };
+  const opts = {
+    include,
+    order: [['createdAt', 'DESC']],
+    limit,
+    offset
+  };
+  if (publishedOn) {
```

Comment:
Should we separate some of these and extract param parsing / cond setting?

Reply:
Separate in the sense of different API endpoints or in some other way?